app/helpers/assessment_query_helper.rb

```diff
@@ -69,35 +69,77 @@ def paginate(assessments, entries, page)
 
   # Formats assessments to be displayed on the frontend.
   def format_for_frontend(assessments)
```

Comment:
I'm seeing 5 different SQL queries being sent to the database in this method. Is there a way that we could refactor this to use fewer DB calls with joins, eager_loads, etc?

Reply:
so after spending a couple hours playing around with this, I think the way it's written now is better than trying to do any kind of joins because that seems to be slower. I tried doing stuff like this to get conditions and symptoms at the same time, but it's either about the same or slower in terms of speed:

```
threshold_symptoms = Symptom.joins('JOIN conditions reported_conditions ON symptoms.condition_id = reported_conditions.id')
                            .joins('JOIN conditions threshold_conditions ON threshold_conditions.threshold_condition_hash = reported_conditions.threshold_condition_hash')
                            .joins('JOIN assessments ON reported_conditions.assessment_id = assessments.id')
                            .where('assessments.id IN (?)', assessment_ids)
                            .where('threshold_conditions.type = ?', 'ThresholdCondition')
                            .select('threshold_conditions.threshold_condition_hash', 'symptoms.name', 'symptoms.label', 'symptoms.type', 'symptoms.required', 'symptoms.threshold_operator', 'symptoms.bool_value', 'symptoms.int_value', 'symptoms.float_value')
                            .group_by(&:threshold_condition_hash)
                            .transform_values { |symptoms| Hash[symptoms.map { |symptom| [symptom[:name], symptom] }] }
```

Before:
```
(0.5ms)  SELECT `conditions`.`id`, `conditions`.`threshold_condition_hash` FROM `conditions` WHERE `conditions`.`type` = 'ThresholdCondition' AND `conditions`.`threshold_condition_hash` IN (SELECT `conditions`.`threshold_condition_hash` FROM `conditions` WHERE `conditions`.`type` = 'ReportedCondition' AND `conditions`.`assessment_id` IN (2691, 1592, 1443, 747, 172, 134, 49))
  ↳ app/helpers/assessment_query_helper.rb:81:in `format_for_frontend'
Symptom Load (0.4ms)  SELECT `symptoms`.`condition_id`, `symptoms`.`name`, `symptoms`.`label`, `symptoms`.`type`, `symptoms`.`required`, `symptoms`.`threshold_operator`, `symptoms`.`bool_value`, `symptoms`.`int_value`, `symptoms`.`float_value` FROM `symptoms` WHERE `symptoms`.`condition_id` IN (13, 8)
  ↳ app/helpers/assessment_query_helper.rb:86:in `format_for_frontend'
```

After:
```
Symptom Load (1.3ms)  SELECT `threshold_conditions`.`threshold_condition_hash`, `symptoms`.`name`, `symptoms`.`label`, `symptoms`.`type`, `symptoms`.`required`, `symptoms`.`threshold_operator`, `symptoms`.`bool_value`, `symptoms`.`int_value`, `symptoms`.`float_value` FROM `symptoms` JOIN conditions reported_conditions ON symptoms.condition_id = reported_conditions.id JOIN conditions threshold_conditions ON threshold_conditions.threshold_condition_hash = reported_conditions.threshold_condition_hash JOIN assessments ON reported_conditions.assessment_id = assessments.id WHERE (assessments.id IN (2691,1592,1443,747,172,134,49)) AND (threshold_conditions.type = 'ThresholdCondition')
  ↳ app/helpers/assessment_query_helper.rb:97:in `group_by'
```